### DISCUSSION TITLE
Improve maintenance by phasing out / improving modules

### DISCUSSION THREAD

**Discussion by chahank** [CLIMADA-project/climada_petals#99] [Ideas]
Certain modules, in particular in petals have been produced before a proper review process was introduced. Thus, certain code bits do not follow the climada conventions and are hard to maintain. This becomes particularly important for the ground-up improvements that are planned such as the consolidation of data structures (see eg. https://github.com/CLIMADA-project/climada_python/pull/787).

Hence, I would like to open a discussion on whether certain modules could be either brought up to the standards of the rest of the code or phased out. A potential list of such modules are:

- [black_marble.py](https://github.com/CLIMADA-project/climada_petals/blob/main/climada_petals/entity/exposures/black_marble.py)
- [low_flow.py](https://github.com/CLIMADA-project/climada_petals/blob/main/climada_petals/hazard/low_flow.py)
- [drought.py](https://github.com/CLIMADA-project/climada_petals/blob/main/climada_petals/hazard/drought.py)
- [emulator](https://github.com/CLIMADA-project/climada_petals/tree/main/climada_petals/hazard/emulator)

**Comment by peanutfun** [comment]
We decided in today's Climada Senate:
* Modules in Petals will require a "Person in charge". This person is responsible for updating the module to new releases of Climada Core, and for fixing bugs. The main developers only provide support.
   * Whenever blockers arise, especially before a release, the Person in Charge has 2 weeks to resolve the issues or give a detailed plan on how the issues will be resolved in the subsequent 2 weeks. If the blockers cannot be resolved, unless there is a clear justification why resolving issues might take longer, the module will be **removed** from the next Petals release.
* NEW modules in Petals will require a citable, technical document (in PDF format) with DOI. This document must contain a high-level overview of the module/model, and describe modeling choices like input data, parameters, etc. If a journal paper or similar document was published alongside the model, this publication can be cited if it contains the appropriate information. This will become a requirement for merging new modules.
* A new page in the Petals documentation will state the technical documents and person in charge for each Petals module. It will probably look like the Core [Citation Guide](https://climada-python.readthedocs.io/en/latest/misc/citation.html).
